.github/workflows/build.yml

```diff
@@ -81,11 +81,31 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install package
-        uses: ./.github/actions/install-cpu
+        uses: ./.github/actions/install
         with:
           editable: true
       - name: Run unit tests and generate coverage report
         run: |
           set -o pipefail  # To propagate exit code from pytest
           coverage run --source=graphnet -m pytest tests/ --ignore=tests/data/
           coverage report -m
+
+  build-macos:
+    name: Unit tests - macOS
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
```

Comment:
Does the mac implementation only support python 3.8?
Or would something like
```
strategy:
      matrix:
        python-version: [3.7, 3.8, 3.9, '3.10']
```
still work?

Reply:
That should work just as well too! The reason I opted for a single, fixed version is:

* We only test for support for python versions 3.8 through 3.10, so I figures it was good to pick one, fixed version to test against rather than using whatever comes with `macos-latest` by default (which could change over time).
* Running on the macOS image is [expensive!](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers) On the free tier, we have a fixed number of free GitHub Actions minutes and running on macOS "eats" those minutes x10 faster than running on Ubuntu. Since we're testing for different python versions in another job (the matrix one) I figured it was fine to have this job dedicated just to testing the OS-specific differences. (Assuming, then, that the python version results from the matrix build job would carry over to macOS as well.)

Reply:
These are very good reasons, I'll consider them resolved!

Reply:
Just FYI, my understanding is that those 2000 minutes are for private repos; GitHub Actions on public repos are not limited. But I agree that testing once for OS-specific differences should be enough.